### ISSUE TITLE
More PHP 8.2 warnings

### ISSUE THREAD

**Issue by Stefan-MyBB** [mybb/mybb#4737]
- [ ] Warning Undefined array key "thumbnail" - Line: 89 - File: attachment.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined array key "dateline" - Line: 1000 - File: inc/functions_post.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined array key 10 - Line: 1094 - File: forumdisplay.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 134 - File: archive/global.php PHP 8.2.8 (Linux) [same issue in lines 116 and 125]
- [ ] Warning Trying to access array offset on value of type bool - Line: 83 - File: archive/index.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type bool - Line: 603 - File: inc/class_session.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined array key 1 - Line: 36 - File: inc/functions_online.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 194 - File: inc/class_datacache.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined array key "" - Line: 1015 - File: private.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 367 - File: showthread.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type bool - Line: 73 - File: attachment.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type bool - Line: 75 - File: attachment.php PHP 8.2.8 (Linux)
- [ ] Warning Array to string conversion : eval()'d code:48 - Line: 1519 - File: newreply.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $awaysection : eval()'d code:88 - Line: 756 - File: usercp.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $thumblist - Line: 1033 - File: inc/functions_post.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 337 - File: showthread.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $words - Line: 7102 - File: inc/functions.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $tcount - Line: 717- File: portal.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $referral_bit : eval()'d code:14 - Line: 531 - File: memberlist.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined variable $referral_header : eval()'d code:50 - Line: 548 - File: memberlist.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 3272 - File: inc/functions.php PHP 8.2.8 (Linux)
- [ ] Warning Undefined array key "activity" - Line: 355 - File: inc/functions_online.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 273 - File: showthread.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 1459 - File: member.php PHP 8.2.8 (Linux)
- [ ] Warning Trying to access array offset on value of type null - Line: 1280 - File: member.php PHP 8.2.8 (Linux)

**Comment by spikyatlinux** [comment]
If board is deactivated you get the following warning under PHP 8.2

`Warning [2] Undefined array key "index.php" - Line: 1105 - File: global.php PHP 8.1.20 (Linux)`

The complete warning is 

```
/inc/class_error.php 	153 	errorHandler->error
/global.php 	1105 	errorHandler->error_callback
/index.php 	18 	require_once
```

**Comment by Stefan-MyBB** [comment]
@spikyatlinux You are neither using PHP 8.2 nor reporting a valid issue. Line 1105 in global.php looks like this:
```php
 if(
```

**Comment by spikyatlinux** [comment]
> @spikyatlinux You are neither using PHP 8.2 nor reporting a valid issue. Line 1105 in global.php looks like this:
> 
> ```
>  if(
> ```
I'm sorry but under PHP 7.4 there wasn't an error like this.
I've got the same error on my other system with PHP 8.2.
Best regards 

```
Warning [2] Undefined array key "index.php" - Line: 1105 - File: global.php PHP 8.2.10 (Linux)
File 	Line 	Function
/inc/class_error.php 	153 	errorHandler->error
/global.php 	1105 	errorHandler->error_callback
/index.php 	18 	require_once
```

**Comment by Stefan-MyBB** [comment]
@spikyatlinux It seems this issue you are reporting is already fixed since MyBB 1.8.31: https://github.com/mybb/mybb/commit/f98cfd931e5c3c99818d5d5c215b9ea0e9f9ee24#diff-7b7d4d5231d83ea738098e31c83ba44004c829ccf7c74ac947ab02b054fe39ceL1105

**Comment by spikyatlinux** [comment]
@Stefan-MyBB that's strange I´m running latest 1.8.36 version, I´ve manually merged the changes, now the error has gone. Thanks for helping.
